app/core/Vault.js

```diff
@@ -94,16 +87,22 @@ export const recreateVaultWithNewPassword = async (
   const hdKeyring = KeyringController.state.keyrings[0];
   const existingAccountCount = hdKeyring.accounts.length;
 
-  const ledgerKeyring = await getLedgerKeyring();
-  const qrKeyring = (
-    await KeyringController.getKeyringsByType(KeyringTypes.qr)
-  )[0];
+  const serializedLedgerKeyring = hasKeyringType(KeyringTypes.ledger)
+    ? await getSerializedKeyring(KeyringTypes.ledger)
+    : undefined;
+  const serializedQrKeyring = hasKeyringType(KeyringTypes.qr)
+    ? await getSerializedKeyring(KeyringTypes.qr)
+    : undefined;
```

Comment:
This function is most likely failing to recreate hardware keyrings when changing password because `hasKeyringType` always returns false, because we should pass the KeyringController instance state:

```suggestion
  const serializedLedgerKeyring = hasKeyringType(
    KeyringController.state,
    KeyringTypes.ledger,
  )
    ? await getSerializedKeyring(KeyringTypes.ledger)
    : undefined;
  const serializedQrKeyring = hasKeyringType(
    KeyringController.state,
    KeyringTypes.qr,
  )
    ? await getSerializedKeyring(KeyringTypes.qr)
    : undefined;
```

Reply:
Alternatively, we can mimic `getSerializedKeyring` by selecting the `KeyringController.state` in `hasKeyringType` itself